docs/ops/movement/Roll_7.md

```diff
@@ -0,0 +1,132 @@
+## Roll <a name="Roll"></a> {#openvino_docs_ops_movement_Roll_7}
+
+**Versioned name**: *Roll-7*
+
+**Category**: data movement operation
+
+**Short description**: *Roll* operation shifts elements of a tensor along specified axes.
+
+**Detailed description**: *Roll* produces a tensor with the same shape as the first input tensor and with elements shifted along dimensions specified in *axes* tensor. The shift size is specified in *shift* input tensor. Elements that are shifted beyond the last position will be added in the same order starting from the first position.
+
+Example 1. *Roll* output with `shift` = 1, `axes` = 0:
+
+``` 
+data    = [[ 1,  2,  3],
+        [ 4,  5,  6],
+        [ 7,  8,  9],
+        [10, 11, 12]]
+output  = [[10, 11, 12],
+        [ 1,  2,  3],
+        [ 4,  5,  6],
+        [ 7,  8,  9]]
+```
+
+Example 2. *Roll* output with `shift` = [-1, 2], `axes` = [0, 1]:
+
+``` 
+data    = [[ 1,  2,  3],
+        [ 4,  5,  6],
+        [ 7,  8,  9],
+        [10, 11, 12]]
+output  = [[ 5,  6,  4],
+        [ 8,  9,  7],
+        [11, 12, 10],
+        [ 2,  3,  1]]
+```
+
+Example 3. *Roll* output with `shift` = [1, 2, 1], `axes` = [0, 1, 0]:
+
+``` 
+data    = [[ 1,  2,  3],
+        [ 4,  5,  6],
+        [ 7,  8,  9],
+        [10, 11, 12]]
+output  = [ 8,  9,  7],
+        [11, 12, 10],
+        [ 2,  3,  1],
+        [ 5,  6,  4]]
+```
+
+**Attributes**
+
+No attributes available.
+
+**Inputs**:
+
+*   **1**: `data` the tensor of type *T*. **Required.**
+
+*   **2**: `shift` scalar or 1D tensor of type *T_IND_1* which specifies the number of places by which the elements of `data` tensor are shifted. If `shift` is a scalar, each dimension specified in `axes` tensor are rolled by same `shift` value. If `shift` is 1D tensor, `axes` must be a 1D tensor of the same size, and each dimension from `axes` tensor are rolled by the corresponding value from `shift` tensor. If the value of `shift` is positive, elements are shifted positively (towards larger indices). Otherwise elements are shifted negatively (towards smaller indices). **Required.**
```

Comment:
```suggestion
*   **2**: a `shift` scalar or 1D tensor of type *T_IND_1*. Specifies the number of places by which the elements of the `data` tensor are shifted. If `shift` is a scalar, each dimension specified in the `axes` tensor are rolled by the same `shift` value. If `shift` is a 1D tensor, `axes` must be a 1D tensor of the same size, and each dimension from `axes` tensor are rolled by the corresponding value from the `shift` tensor. If the value of `shift` is positive, elements are shifted positively (towards larger indices). Otherwise, elements are shifted negatively (towards smaller indices). **Required.**
```